### PR TITLE
CI: use separate workflow for deployment

### PR DESCRIPTION
**Pull request by sebproell**
Try to make the overview of actions a bit more intuitive by not showing the skipped deployment in PRs.

### CODE REVIEW

**Comment by sebproell** [comment]
This achieves what I wanted in a PR. Should we merge and see if it does the correct thing on `main`?

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+on:
+  workflow_run:
+    workflows: [documentation.yml]
+    types: [completed]
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
```
Comment (amgebauer) [inline .github/workflows/deploy.yml:12]:
```suggestion
jobs:
 on-success:
```

I think we need this to ensure that we only deploy if build of doxygen and readthedocs was successful

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: deploy
+
+on:
+  workflow_run:
+    workflows: [documentation.yml]
```
Comment (ppraegla) [inline .github/workflows/deploy.yml:5]: Is this the correct syntax or does it need to be the workflow name?
```suggestion
    workflows: [documentation]
```

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (sebproell) [reply .github/workflows/deploy.yml:12]: I agree that we might need such a check but isn't `on-success` just the name of the job?

Comment (amgebauer) [reply .github/workflows/deploy.yml:12]: true, I read too fast over the docs

Comment (sebproell) [reply .github/workflows/deploy.yml:12]: Condition added.

**Comment by amgebauer** [comment]
@sebproell merge?

**Pull request merged**